django/contrib/gis/gdal/geometries.py

```diff
@@ -268,6 +268,20 @@ def extent(self):
         "Return the envelope as a 4-tuple, instead of as an Envelope object."
         return self.envelope.tuple
 
+    @property
+    def is_3d(self):
+        """Return True if the geometry has Z coordinates."""
+        return capi.is_3d(self.ptr)
+
+    def set_3d(self, value):
+        """Set if this geometry has Z coordinates."""
+        if value is True:
+            capi.set_3d(self.ptr, 1)
+        elif value is False:
+            capi.set_3d(self.ptr, 0)
+        else:
+            raise ValueError(f"Input to 'set_3d' must be a boolean, got {value!r}.")
```

Comment:
Can we also switch to these methods when possible? e.g. `OGRGeometry.z`, `OGRGeometry.tuple`, or `verify_geom()` as you did in #17696.